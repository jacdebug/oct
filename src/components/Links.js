import React from 'react';
import MainNavigationLink from './MainNavigationLink';

const Links = ({ links, bright }) =>
  <div>
    {links.map((link, index) =>
      <MainNavigationLink bright={bright} to={link.url} key={index}>
        {' '} {link.name} {' '}
      </MainNavigationLink>
    )}
  </div>;

export default Links;
