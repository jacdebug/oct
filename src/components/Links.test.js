import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Links from './Links';

const linksTestArray = [
  {
    name: 'link1',
    url: '/',
  },
  {
    name: 'link3',
    url: '/',
  },
];

it('renders without crashing', () => {
  shallow(<Links links={linksTestArray} />);
});

it('renders a snapshot', () => {
  const tree = shallow(<Links links={linksTestArray} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders list of links', () => {
  const wrapper = shallow(<Links links={linksTestArray} />);
  expect(wrapper.find('Styled(Component)')).toHaveLength(2);
});
