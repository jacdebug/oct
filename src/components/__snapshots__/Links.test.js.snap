// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <Styled(Component)
    to="/"
  >
     
     
    link1
     
     
  </Styled(Component)>
  <Styled(Component)
    to="/"
  >
     
     
    link3
     
     
  </Styled(Component)>
</div>
`;
